### PR TITLE
[SPARK-38496][PYTHON][TESTS] Add unit tests for map functions

### PR DESCRIPTION
**Pull request by khalidmammadov**
### What changes were proposed in this pull request?
Adding missing Unit tests

### Why are the changes needed?

Increase test coverage for PySpark  API

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
`python/run-tests`

### CODE REVIEW

**Comment by khalidmammadov** [comment]
@HyukjinKwon @itholic please review

File: python/pyspark/sql/tests/test_functions.py [hunk 1]
```diff
@@ -1072,6 +1072,44 @@ def test_binary_math_function(self):
         for a, e in zip(df.first(), expected):
             self.assertAlmostEqual(a, e, 5)
 
+    def test_map_functions(self):
+        from pyspark.sql import functions as F
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_functions.py:1076]: Let's move this import to the top, and add a comment like:

```python
def test_case(self):
    # SPARK-12345: a short description of the test
    ...
```

See also https://spark.apache.org/contributing.html

**Review by HyukjinKwon — COMMENTED** [review]
Looks pretty good otherwise.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Review by HyukjinKwon — APPROVED** [review]

**Pull request closed without merging**